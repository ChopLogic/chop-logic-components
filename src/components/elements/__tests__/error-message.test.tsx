import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import ChopLogicErrorMessage from '../error-message/ErrorMessage';

describe('ChopLogicErrorMessage component', () => {
  const testProps = {
    errorId: 'test-error-id',
    visible: true,
    message: 'Incorrect input',
    className: 'custom-error',
  };

  it('should render correctly', () => {
    const { asFragment } = render(<ChopLogicErrorMessage {...testProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
