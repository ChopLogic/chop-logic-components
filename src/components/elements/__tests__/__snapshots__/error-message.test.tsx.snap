// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChopLogicErrorMessage component > should render correctly 1`] = `
<DocumentFragment>
  <span
    class="_message_a256d5 custom-error _visible_a256d5"
    id="test-error-id"
  >
    Incorrect input
  </span>
</DocumentFragment>
`;
